client/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import FocusLock from "react-focus-lock";
+
+import "./Sidebar.scss";
+
+import { CSSTransition } from "react-transition-group";
+
+import { FloatingButton } from "src/components/FloatingButton/FloatingButton";
+
+import { IconX } from "src/icons/icons";
+
+interface SidebarProps {
+  is_open: boolean;
+  callback: (value: boolean) => void;
+  children: React.ReactElement;
+  button_text: string;
+  title_text: string;
+}
+
+export class Sidebar extends React.Component<SidebarProps> {
+  title = React.createRef<HTMLDivElement>();
+  constructor(props: SidebarProps) {
+    super(props);
+  }
+
+  handleKeyPress(e: React.KeyboardEvent<HTMLSpanElement>) {
+    if (e.key === "Enter") {
+      this.close();
+    }
+  }
+
+  close() {
+    this.props.callback(false);
+  }
+
+  render() {
+    const { is_open, children, button_text, title_text } = this.props;
+    return (
+      <div>
+        <FocusLock>
+          <CSSTransition
+            in={is_open}
+            timeout={1000}
+            classNames="slide"
+            appear
+            mountOnEnter
+            unmountOnExit
+            onEnter={() => this.title.current?.focus()}
+          >
+            <div className={"sidebar__wrapper"}>
+              <aside className="sidebar">
+                <div className={"sidebar__icon-wrapper"}>
+                  <span
+                    role="button"
+                    className="sidebar__close-button"
+                    onClick={() => this.props.callback(false)}
+                    onKeyDown={(e) => this.handleKeyPress(e)}
+                    tabIndex={0}
+                  >
+                    <IconX width="25px" color="white" alternate_color={false} />
+                  </span>
+                </div>
+                <div className={"sidebar__title"}>
+                  <h1
+                    ref={this.title}
+                    /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
+                    tabIndex={0}
+                  >
+                    {title_text}
+                  </h1>
+                </div>
+                {children}
+              </aside>
+            </div>
+          </CSSTransition>
+        </FocusLock>
+        <FloatingButton
+          button_text={button_text}
+          showWithScroll={false}
+          handleClick={() => this.props.callback(true)}
+          tabIndex={0}
+        />
```

Comment:
Is a floating button inherent to a sidebar? We might not have a use case, right now at least, for a sidebar that is opened by something other than a floating button, but I don't think the base `Sidebar` component should package them both. 

PR's old enough that it's getting hard to keep track of my own past advice, haha, but I believe you may have done this in response to my comments on the previous `SidebarButton`. I was saying there that something called `SidebarButton` should include a sidebar in addition to a button, and probably also manage it's own state. I also said, though, that I didn't think we needed that component to exist right now. I didn't mean to suggest that it be pushed down in to `Sidebar` itself though, I was thinking that rendering the `FloatingButton` that controls the glossary sidebar should be `GlossarySidebarController`'s responsibility.

A common pattern you'll see with our components is to exist in two versions. One stateless version that's just presentation and as much inherent behaviour as can be encapsulated without needing to own too much state, and one version with state, and possibly additional controls, that is a fully usable implementation. We sometimes call the second type "auto" as opposed to "stateless", but I don't think I've always held consistent on that. The point of this approach is to have both a ready to use/reference implementation for the most default use case, as well as a version that gives up more control/responsibility to the consumer for non-standard use cases.

Circling back, the more flexible, stateless, `Sidebar` shouldn't render its own control. That should be something the consumer can pick/is responsible for. If you want to/anticipate us using an auto version, then you can make that a separate module that imports the stateless `Sidebar` and packages it with a `FloatingButton` (and self-managed open/closed state). Since the glossary sidebar use case needs special behaviour, opening on glossary link click, it won't be using the auto version either way.

Reply:
While I'm at it though, there _is_ some code in `GlossarySideBar` controller that should belong to even the stateless `Sidebar`, I believe. I think we consider closing on click outside to be inherent to a sidebar, right? The stateless version should probably manage window click events and call it's closing callback as necessary. Pushing that inherent behaviour down to this level will save us duplicating it across all sidebars going forward.

If we decide there's a sidebar that _shouldn't_ have that behaviour in the future, then just add a prop to disable it. Still better for it to be something encapsulated at the lowest level.